kanvas/src/androidMain/kotlin/PaintCache.kt

```diff
@@ -0,0 +1,13 @@
+package com.juul.krayon.kanvas
+
+import android.content.Context
+import android.graphics.Paint as AndroidPaint
+
+public class PaintCache(
+    private val context: Context,
+    private val backing: MutableMap<Paint, AndroidPaint> = HashMap(),
+) {
+    public operator fun get(paint: Paint): AndroidPaint = synchronized(backing) {
```

Comment:
It's a bit odd to by synchronizing on a mutable map that is provided from outside of the class. If the caller modifies the map, `synchronized` won't do anything to ensure concurrency.

If it's never necessary to modify `backing` outside of `PaintCache`, it would be better to initialize `PaintCache` with an immutable map.

```kotlin
public class PaintCache(
    private val context: Context,
    initial: Map<Paint, AndroidPaint> = emptyMap(),
) {
    private val backing = initial.toMutableMap()
  
    // ...
}